gm8emulator/src/game/recording/game_window.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{
+    imgui_utils::*,
+    game::{
+        Renderer,
+        recording::{
+            instance_report::InstanceReport,
+            set_mouse_dialog::{SetMouseDialog, MouseDialogResult},
+            window::{Window, EmulatorContext},
+        },
+    },
+};
+
+// for imgui callback
+struct GameViewData {
+    renderer: *mut Renderer,
+    x: i32,
+    y: i32,
+    w: u32,
+    h: u32,
+}
+
+pub struct GameWindow {
+    callback_data: GameViewData,
+    context_menu_options: Option<Vec<(String, i32)>>,
+    mouse_dialog: SetMouseDialog,
+    set_screencover_focus: bool,
+}
+
+// Game window
+impl Window for GameWindow {
+    fn name(&self) -> String {
+        "Game".to_owned()
+    }
+
+    fn show_window(&mut self, info: &mut EmulatorContext) {
+        if *info.game_running {
+            self.display_window(info);
+        } else {
+            *info.setting_mouse_pos = false;
+        }
+    }
+
+    fn is_open(&self) -> bool { true }
+
+    fn show_context_menu(&mut self, info: &mut EmulatorContext) -> bool {
+        self.display_context_menu(info)
+    }
+}
+
+impl GameWindow {
+    pub fn new() -> GameWindow {
+        GameWindow {
+            callback_data: GameViewData {
+                w: 0,
+                h: 0,
+                x: 0,
+                y: 0,
+                renderer: std::ptr::null_mut(),
+            },
+            context_menu_options: None,
+            mouse_dialog: SetMouseDialog::new(),
+            set_screencover_focus: true,
+        }
+    }
+
+    fn display_window(&mut self, info: &mut EmulatorContext) {
+        if *info.setting_mouse_pos {
+            let cover = info.frame.begin_screen_cover(self.set_screencover_focus);
+            let screencover_focused = info.frame.is_window_focused();
+            cover.end();
+
+            if self.set_screencover_focus {
+                info.frame.set_next_window_focus();
+            }
+
+            if info.config.set_mouse_using_textbox {
+                self.mouse_dialog.init_if_closed(info.new_mouse_pos.unwrap_or_else(|| {
+                    // Take the mouse x/y from the previous frame if we don't have one set for this frame
+                    // Technically it would be better if it took the current frame if it existed, the previous if not and (0, 0) if neither exist but if we're in a state where the current frame already exists we're most likely in read-only anyway and can't edit the mouse using this window. (TODO: allow editing current frame using this window too?)
```

Comment:
Please break this into multiple lines (120 is our line length limit atm).
Also, TODOs must always start on a separate line.